checker_test.go

```diff
@@ -0,0 +1,384 @@
+	got:                   map[string]int{"answer": 42},
+	args:                  []interface{}{".*"},
+	expectedCheckFailure:  "expected a function, got map[string]int instead",
+	expectedNegateFailure: "expected a function, got map[string]int instead",
+}, {
+	about:                 "PanicMatches: not a proper function",
+	checker:               qt.PanicMatches,
+	got:                   func(int) { panic("error: bad wolf") },
+	args:                  []interface{}{".*"},
+	expectedCheckFailure:  "expected a function accepting no arguments, got func(int) instead",
+	expectedNegateFailure: "expected a function accepting no arguments, got func(int) instead",
+}, {
+	about:   "PanicMatches: function returning something",
+	checker: qt.PanicMatches,
+	got:     func() error { panic("error: bad wolf") },
+	args:    []interface{}{".*"},
+	expectedNegateFailure: `there was a panic matching ".*"`,
+}, {
+	about:                "PanicMatches: no panic",
+	checker:              qt.PanicMatches,
+	got:                  func() {},
+	args:                 []interface{}{".*"},
+	expectedCheckFailure: "the function did not panic",
+}, {
+	about:   "IsNil: nil",
+	checker: qt.IsNil,
+	got:     nil,
+	expectedNegateFailure: "the value is nil, but should not",
+}, {
+	about:   "IsNil: nil struct",
+	checker: qt.IsNil,
+	got:     (*struct{})(nil),
+	expectedNegateFailure: "the value is nil, but should not",
+}, {
+	about:   "IsNil: nil func",
+	checker: qt.IsNil,
+	got:     (func())(nil),
+	expectedNegateFailure: "the value is nil, but should not",
+}, {
+	about:   "IsNil: nil map",
+	checker: qt.IsNil,
+	got:     (map[string]string)(nil),
+	expectedNegateFailure: "the value is nil, but should not",
+}, {
+	about:   "IsNil: nil slice",
+	checker: qt.IsNil,
+	got:     ([]int)(nil),
+	expectedNegateFailure: "the value is nil, but should not",
+}, {
+	about:                "IsNil: not nil",
+	checker:              qt.IsNil,
+	got:                  42,
+	expectedCheckFailure: "42 is not nil",
+}, {
+	about:   "Not: success",
+	checker: qt.Not(qt.IsNil),
+	got:     42,
+	expectedNegateFailure: "42 is not nil",
+}, {
+	about:                "Not: failure",
+	checker:              qt.Not(qt.IsNil),
+	got:                  nil,
+	expectedCheckFailure: "the value is nil, but should not",
+}}
+
+func TestCheck(t *testing.T) {
+	for _, test := range checkerTests {
+		t.Run(test.about, func(t *testing.T) {
+			err := test.checker.Check(test.got, test.args)
+			if test.expectedCheckFailure != "" {
+				assertErrHasPrefix(t, err, test.expectedCheckFailure)
+				return
+			}
+			assertErrIsNil(t, err)
+		})
+	}
+}
+
+func TestNegate(t *testing.T) {
+	for _, test := range checkerTests {
+		t.Run(test.about, func(t *testing.T) {
+			err := test.checker.Negate(test.got, test.args)
+			if test.expectedNegateFailure != "" {
+				assertErrHasPrefix(t, err, test.expectedNegateFailure)
+				return
+			}
+			assertErrIsNil(t, err)
+		})
+	}
+}
+
+func TestCCheck(t *testing.T) {
```

Comment:
I don't get what this is testing that's different from TestCheck. Please use a more descriptive test name (same for TestCNegate)

Reply:
Effectively a redundant test, removed.